docker/Dockerfile-refact-debug

```diff
@@ -0,0 +1,34 @@
+# This dockerfile can be used to compile refact-lsp for development purposes, 
+# for example, to get refact-lsp to bind into docker containers to start threads in containers
+
+FROM ubuntu:24.04
+
+RUN apt-get update && apt-get install -y \
+    build-essential \
+    curl \
+    git \
+    libssl-dev \
+    pkg-config \
+    protobuf-compiler \
+    && rm -rf /var/lib/apt/lists/*
+
+RUN curl https://sh.rustup.rs -sSf | sh -s -- -y
+ENV PATH=/root/.cargo/bin:$PATH
+
+WORKDIR /refact-lsp
+
+COPY Cargo.toml build.rs ./
+
+RUN mkdir src && echo 'fn main() { println!("Dummy main to satisfy Cargo"); }' > src/main.rs
+
+RUN cargo fetch
+RUN cargo check
+
+RUN rm -rf src
+
+COPY . .
+
+RUN cargo build
+
+RUN mkdir -p /output && cp target/debug/refact-lsp /output/
```

Comment:
hmm why /output ?

Reply:
The Dockerfile is primarily for building refact-lsp for development. We’ll extract the binary to the local machine. The /output directory is a bit generic, but I've seen it a few times; if you have a clearer name suggestion, we can rename it.